kgforge/core/archetypes/read_store.py

```diff
@@ -0,0 +1,222 @@
+            follow: str,
+            path: str,
+            overwrite: bool,
+            cross_bucket: bool,
+            content_type: str = None
+    ) -> None:
+        # path: DirPath.
+        urls = []
+        store_metadata = []
+        to_download = [data] if isinstance(data, Resource) else data
+        for d in to_download:
+            collected_values = collect_values(d, follow, DownloadingError)
+            urls.extend(collected_values)
+            store_metadata.extend(
+                [d._store_metadata for _ in range(len(collected_values))]
+            )
+        if len(urls) == 0:
+            raise DownloadingError(
+                f"path to follow '{follow}' was not found in any provided resource."
+            )
+        dirpath = Path(path)
+        dirpath.mkdir(parents=True, exist_ok=True)
+        timestamp = time.strftime("%Y%m%d%H%M%S")
+        filepaths = []
+        buckets = []
+        download_urls = []
+        download_store_metadata = []
+        for i, x in enumerate(urls):
+            x_download_url, x_bucket = self._prepare_download_one(x, store_metadata[i],
+                                                                  cross_bucket)
+            filename, store_content_type = self._retrieve_filename(x_download_url)
+            if not content_type or (content_type and store_content_type == content_type):
+                filepath = dirpath / filename
+                if not overwrite and filepath.exists():
+                    filepaths.append(f"{filepath}.{timestamp}")
+                else:
+                    filepaths.append(str(filepath))
+                download_urls.append(x_download_url)
+                buckets.append(x_bucket)
+                download_store_metadata.append(store_metadata[i])
+        if len(download_urls) > 1:
+            self._download_many(download_urls, filepaths, download_store_metadata, cross_bucket,
+                                content_type, buckets)
+        elif len(download_urls) == 1:
+            self._download_one(download_urls[0], filepaths[0], download_store_metadata[0],
+                               cross_bucket, content_type, buckets[0])
+        else:
+            raise DownloadingError(
+                f"No resource with content_type {content_type} was found when following the resource path '{follow}'."
+            )
+
+    def _download_many(
+            self,
+            urls: List[str],
+            paths: List[str],
+            store_metadata: Optional[List[DictWrapper]],
+            cross_bucket: bool,
+            content_type: str,
+            buckets: List[str]
+    ) -> None:
+        # paths: List[FilePath].
+        # Bulk downloading could be optimized by overriding this method in the specialization.
+        # POLICY Should follow self._download_one() policies.
+        for url, path, store_m in zip(urls, paths, store_metadata):
+            self._download_one(url, path, store_m, cross_bucket, content_type)
```

Comment:
Probably it a problem that was already there but the buckets argument is not used, and the _download_one call is missing a bucket parameter

Reply:
good catch